src/sources/play-dl-source/strategies/spotify-strategy.ts

```diff
@@ -0,0 +1,45 @@
+import play, { SpotifyPlaylist, SpotifyTrack } from 'play-dl';
+
+import { ResultAudioSearch, StreamInfo } from '@/sources/source-stream';
+
+import { PlayDlSourceStream } from '../play-dl-source';
+import { IStrategy } from './strategy';
+
+export class SpotifyTrackStrategy implements IStrategy {
+  constructor(private playDlSourceStream: PlayDlSourceStream) {}
+
+  async getStreamInfo(url: string): Promise<StreamInfo[]> {
+    const spotifyInfo = (await play.spotify(url.trim())) as SpotifyTrack;
+
+    const searched = (await this.playDlSourceStream.search(
+      `${spotifyInfo.name} - ${spotifyInfo?.artists[0]?.name}`,
```

Comment:
also, you can use shift() here

Reply:
resolved: [5252756](https://github.com/Marli-Marlete/marli-music/pull/21/commits/5252756b4ba49d970306720a3bb9a66014cda2c6)